File: backend/src/main/java/com/hardware/erp/product/service/BrandService.java
package com.hardware.erp.product.service;

import com.hardware.erp.product.dto.BrandRequest;
import com.hardware.erp.product.dto.BrandResponse;

import java.util.List;

public interface BrandService {

    List<BrandResponse> findAll();

    BrandResponse get(Long id);

    BrandResponse create(BrandRequest request);

    BrandResponse update(Long id, BrandRequest request);

    void delete(Long id);
}

File: frontend/src/modules/product/services/brandService.ts
import { apiDelete, apiGet, apiPost, apiPut } from '@/services/apiClient';
import type { BrandRequest, BrandResponse } from '../types';

/** Backend: product/controller/BrandController.java */
export const brandService = {
  list: () => apiGet<BrandResponse[]>('/v1/brands'),

  get: (id: number) => apiGet<BrandResponse>(`/v1/brands/${id}`),

  create: (body: BrandRequest) => apiPost<BrandResponse>('/v1/brands', body),

  update: (id: number, body: BrandRequest) => apiPut<BrandResponse>(`/v1/brands/${id}`, body),

  /** Hard delete. Rejected if any product still references it. */
  remove: (id: number) => apiDelete(`/v1/brands/${id}`),
};
